File: appAndroid/common-compose/src/main/java/com/gchristov/newsfeed/commoncompose/theme/Typography.kt
package com.gchristov.newsfeed.commoncompose.theme

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

@Immutable
data class Typography(
    val title: TextStyle,
    val subtitle: TextStyle,
    val body: TextStyle,
    val bodyBold: TextStyle,
    val caption: TextStyle,
)

internal val LocalTypography = staticCompositionLocalOf {
    Typography(
        title = TextStyle.Default,
        subtitle = TextStyle.Default,
        body = TextStyle.Default,
        bodyBold = TextStyle.Default,
        caption = TextStyle.Default,
    )
}

internal fun typography() = Typography(
    title = Title,
    subtitle = Subtitle,
    body = Body,
    bodyBold = BodyBold,
    caption = Caption,
)

private val Title = TextStyle(fontSize = 24.sp)
private val Subtitle = TextStyle(fontSize = 15.sp)
private val Body = TextStyle(fontSize = 18.sp)
private val BodyBold = Body.copy(fontWeight = FontWeight.Bold)
private val Caption = TextStyle(fontSize = 12.sp)

File: ios/common-swiftui/common-swiftui/theme/Typography.swift
import SwiftUI

public struct Typography {
    public let title: Font
    public let subtitle: Font
    public let body: Font
    public let bodyBold: Font
    public let caption: Font
}

func typography() -> Typography {
    return Typography(
        title: Font.system(size: 24),
        subtitle: Font.system(size: 15),
        body: Font.system(size: 18),
        bodyBold: Font.system(size: 18, weight: .bold),
        caption: Font.system(size: 12)
    )
}
